components/ModFacilitySearchbar.vue

```diff
@@ -0,0 +1,181 @@
+                        selectedFacilityIndex === index ? 'bg-primary-hover text-primary-inverted' : 'opacity-95',
+                    ]"
+                    @click="(event) => { event.preventDefault(); handleListItemClick(facility) }"
+                    @mousedown="(event) => {
+                        // Prevent search input from being blurred
+                        event.preventDefault()
+                    }"
+                    @mousemove="selectedFacilityIndex = index"
+                >
+                    <div class="flex flex-col overflow-x-auto whitespace-nowrap m-3">
+                        <span class="text-xs">{{ facility.id }}</span>
+                        <span class="">{{ `${facility.nameEn} / ${facility.nameJa} ` }}</span>
+                    </div>
+                    <div class="flex items-center">
+                        <SVGCheckMark
+                            class="h-4 m-3"
+                            :class="[
+                                selectedFacilities.has(facility) ? 'opacity-100' : 'opacity-10',
+                            ]"
+                        />
+                    </div>
+                </li>
+                <li
+                    v-if="!filteredFacilities.length"
+                    class="m-3 cursor-default"
+                >
+                    <!-- TODO: Change i18n key if necessary -->
+                    <span>{{ $t('modFacilitySearchbar.noFacilitiesWereFound') }}</span>
+                </li>
+            </ul>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, onMounted, onUnmounted, ref, watch } from 'vue'
+import SVGCheckMark from '~/assets/icons/check-mark.svg'
+import SVGLookingGlass from '~/assets/icons/looking-glass.svg'
+import { useFacilitiesStore } from '#build/imports'
+import { useFacilitySearchbarStore } from '~/stores/facilitySearchbarStore'
+import type { Facility } from '~/typedefs/gqlTypes'
+
+//The limit for fetching facilities is 400 at the current date (30/11/2024)
+const facilitiesStore = useFacilitiesStore()
+const facilitySearchbarStore = useFacilitySearchbarStore()
+
+const searchInput = ref('')
+const searchInputElement = ref<HTMLInputElement | null>(null)
+const filteredFacilities = ref<Facility[]>([])
+const selectedFacilities = ref(facilitySearchbarStore.selectedFacilities)
+const selectedFacilityIndex = ref(0)
+
+const handleListItemClick = (facility: Facility) => {
+    searchInputElement.value?.focus()
+
+    if (selectedFacilities.value.has(facility)) {
+        selectedFacilities.value.delete(facility)
+        return
+    }
+
+    selectedFacilities.value.add(facility)
+}
+
+const handleSearchInputBlur = () => {
+    searchInput.value = ''
+}
+
+const handleSearchInputEsc = () => {
+    handleSearchInputBlur()
+}
+
+const handleSearchInputArrowUp = (event: KeyboardEvent) => {
+    // Prevent the cursor to be moving inside the search input
+    event.preventDefault()
```

Comment:
I noticed you have prevent default happening here but then also in some of the html event hooks. Let's make it more consistent and move them all to the event handlers like you have here.